packages/axelar-wasm-std/src/voting.rs

```diff
@@ -131,10 +134,49 @@ impl fmt::Display for PollId {
     }
 }
 
+#[cw_serde]
+#[derive(Eq, Hash, Ord, PartialOrd, EnumIter, Valuable)]
+pub enum Vote {
+    SucceededOnChain, // the txn was included on chain, and achieved the intended result
+    FailedOnChain,    // the txn was included on chain, but failed to achieve the intended result
+    NotFound,         // the txn could not be found on chain in any blocks at the time of voting
+}
+
+#[cw_serde]
+pub struct Tallies(BTreeMap<Vote, Uint256>);
+
+impl Default for Tallies {
+    fn default() -> Self {
+        let mut tallies = BTreeMap::new();
+
+        Vote::iter().for_each(|vote| {
+            tallies.insert(vote, Uint256::zero());
+        });
+
+        Self(tallies)
+    }
+}
+
+impl Tallies {
+    pub fn consensus(&self, quorum: Uint256) -> Option<Vote> {
+        self.0.iter().find_map(|(vote, tally)| {
+            if *tally >= quorum {
+                Some(vote.clone())
```

Comment:
So it's possible that multiple types of votes exceed the quorum?

Reply:
Task to enforce majority in thresholds: https://axelarnetwork.atlassian.net/browse/AXE-2832